From Juha.Laiho@iki.fi Thu Sep 13 11:21:39 2007
From: Juha Laiho <Juha.Laiho@iki.fi>
Date: Thu, 13 Sep 2007 21:21:34 +0300 (EEST)
Subject: PCI: i386: Compaq EVO N800c needs PCI bus renumbering
To: gregkh@suse.de
Cc: linux-kernel@vger.kernel.org
Message-ID: <Pine.LNX.4.44.0709132041250.21287-100000@ichaos.ichaos-int>


Force PCI bus renumbering for Compaq EVO N800c laptop, in order to get
the cardbus slot recognised.

Signed-off-by: Juha Laiho <Juha.Laiho@iki.fi>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 arch/x86/pci/common.c |   10 ++++++++++
 1 file changed, 10 insertions(+)

--- a/arch/x86/pci/common.c
+++ b/arch/x86/pci/common.c
@@ -287,6 +287,16 @@ static struct dmi_system_id __devinitdat
 			DMI_MATCH(DMI_PRODUCT_NAME, "ProLiant BL685c G1"),
 		},
 	},
+#ifdef __i386__
+	{
+		.callback = assign_all_busses,
+		.ident = "Compaq EVO N800c",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "Compaq"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "EVO N800c"),
+		},
+	},
+#endif
 	{}
 };
 
